Ical.Net/Evaluation/Evaluator.cs

```diff
@@ -17,33 +17,41 @@ protected void IncrementDate(ref CalDateTime dt, RecurrencePattern pattern, int
         if (interval == 0)
             return;
 
-        var old = dt;
-        switch (pattern.Frequency)
+        try
         {
-            case FrequencyType.Secondly:
-                dt = old.AddSeconds(interval);
-                break;
-            case FrequencyType.Minutely:
-                dt = old.AddMinutes(interval);
-                break;
-            case FrequencyType.Hourly:
-                dt = old.AddHours(interval);
-                break;
-            case FrequencyType.Daily:
-                dt = old.AddDays(interval);
-                break;
-            case FrequencyType.Weekly:
-                dt = DateUtil.AddWeeks(old, interval, pattern.FirstDayOfWeek);
-                break;
-            case FrequencyType.Monthly:
-                dt = old.AddDays(-old.Day + 1).AddMonths(interval);
-                break;
-            case FrequencyType.Yearly:
-                dt = old.AddDays(-old.DayOfYear + 1).AddYears(interval);
-                break;
-            // FIXME: use a more specific exception.
-            default:
-                throw new Exception("FrequencyType.NONE cannot be evaluated. Please specify a FrequencyType before evaluating the recurrence.");
+            var old = dt;
+            switch (pattern.Frequency)
+            {
+                case FrequencyType.Secondly:
+                    dt = old.AddSeconds(interval);
+                    break;
+                case FrequencyType.Minutely:
+                    dt = old.AddMinutes(interval);
+                    break;
+                case FrequencyType.Hourly:
+                    dt = old.AddHours(interval);
+                    break;
+                case FrequencyType.Daily:
+                    dt = old.AddDays(interval);
+                    break;
+                case FrequencyType.Weekly:
+                    dt = DateUtil.AddWeeks(old, interval, pattern.FirstDayOfWeek);
+                    break;
+                case FrequencyType.Monthly:
+                    dt = old.AddDays(-old.Day + 1).AddMonths(interval);
+                    break;
+                case FrequencyType.Yearly:
+                    dt = old.AddDays(-old.DayOfYear + 1).AddYears(interval);
+                    break;
+                // FIXME: use a more specific exception.
```

Comment:
If `EvaluationException` instead of `EvaluationOutOfRangeException` would suffice, we could fix this with e.g.:
```cs
default:
    throw new EvaluationException(
        $"""
         Evaluation aborted: '{pattern.Frequency}' cannot be evaluated.
         Specify a valid FrequencyType before evaluating the recurrence.
         """
    );
```

Reply:
Yes, absolutely. But didn't touch this one, so lets do it in a different PR please. Suggest to avoid multi-line messages though.

Reply:
As we're beyond our time budget all the time anyways, I don't appreciate extending the scope if not necessary. I know, my own review comments often suggest otherwise though.